src/PHPMailer.php

```diff
@@ -3134,47 +3135,57 @@ public function encodeHeader($str, $position = 'text')
         //corrupt messages with headers longer than 65 chars. See #818
         $lengthsub = 'mail' == $this->Mailer ? 13 : 0;
         $maxlen = static::STD_LINE_LENGTH - $lengthsub;
```

Comment:
This assumption may need looking at too - This came up in #1525 - it's OK to have a line longer than 76 chars if it contains no embedded spaces that it can be folded on - we should only force RFC2047 encoding if a line is over 998.
I think we need another check to see if the line has no 8-bit chars and contains an unspaced sequence longer than `MAX_LINE_LENGTH` chars (I guess something like `/\w{MAX_LINE_LENGTH,}/`), then we should force Q-encoding, which will wrap the whole header to `STD_LINE_LENGTH`. This isn't absolutely needed as there's no significant harm in wrapping earlier other than a mild increase in size, but it would be a "nice to have".

There's another bug related to this in #36, in that the header label is not taken into account in the length - that's not been critical since it only becomes a problem at the 998 char limit, but we should perhaps allow for that by wrapping a bit before 998, in a similar way that we try to avoid folding bugs in `mail()`.

So I think here we might rename `$maxlen` as `$foldlen` and set `$maxlen` to `static::MAX_LINE_LENGTH - $lengthsub`, then use them in appropriate places. This will affect test outcomes.

Everything else in here is great!

Reply:
> This assumption may need looking at too

Actually, I just extracted this from the two branches (B and Q) for the sake of deduplication. Or did I miss anything?

> There's another bug related to this in #36, in that the header label is not taken into account in the length 

You're right that #36 (which I [did notice](https://github.com/PHPMailer/PHPMailer/issues/1469#issuecomment-534108894)) could be fixed as part of this PR, but in general I would opt against also fixing that one in this PR (which already does more than just *Us[ing] Q-encode to wrap too long headers*).

I'm not sure fixing #36 is as trivial since it would require touching several places where the encoded header *value* is just appended to the header name (e.g. `Content-Type`, `Content-Disposition`, etc.), like here:

https://github.com/PHPMailer/PHPMailer/blob/3c1c6e03603364ab9ba774b062ca4a5270b90adb/src/PHPMailer.php#L2941-L2954

Or am I missing anything?

Reply:
Indeed – we don't need to *fix* #36 in this PR, but a fudge here will certainly help. If we force Q+folding at say `MAX_LINE_LENGTH - 50` (which isn't quite RFC, but is harmless), to have a problem related to it you would need to have both a) a header label longer than 48 chars and b) a header value > 998 chars with no spaces, which is a very unlikely combo, sufficiently so that we could probably forget about #36.

Reply:
I'm having second thoughts about my last comment. We don't need to worry about lines between 76 and 998 chars long in an SMTP context, but in PHP and sending via `mail()`, we do, because of the bug noted mentioned the source, and on here in #818, where lines longer than 65 chars cause issues. Because of that we may as well not bother allowing long lines to exist and leave the encoding behaviour as it stands in your PR.

Reply:
I committed an intermediate state (c4b7f6a) where I distinguished `$foldlen` and `$maxlen`, but I reverted this to f71240d, deduplicating the `$maxlen` calculation as originally intended (and hopefully making it clearer).